File: backend/src/main/java/pl/buczak/kacper/fleetmanagement/entity/dao/vehicle/CarPart.java
package pl.buczak.kacper.fleetmanagement.entity.dao.vehicle;

import pl.buczak.kacper.fleetmanagement.entity.dao.BaseEntity;
import pl.buczak.kacper.fleetmanagement.util.enums.PartType;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.Size;

/*
    @author Kacper Buczak 
*/
@Entity
@Table(name = "car_part")
public class CarPart extends BaseEntity {

    @Column(nullable = false, name = "part_type")
    private PartType partType;

    @Size(max = 30)
    @Column(nullable = false, name = "name")
    private String name;

    @Size(max = 30)
    @Column(nullable = false, name = "description")
    private String description;

    public CarPart() {
    }

    public PartType getPartType() {
        return partType;
    }

    public void setPartType(PartType partType) {
        this.partType = partType;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: frontend/src/app/shared/model/vehicle/car-part.ts
import {PartType} from '../enums/part-type.enum';


export interface CarPartAttrs {
  id: number;
  partType: PartType;
  name: string;
  description: string;
}

export class CarPart {
  constructor(attrs: Partial<CarPartAttrs> = {}) {
    this.id = attrs.id;
    this.partType = attrs.partType;
    this.description = attrs.description;
    this.name = attrs.name;
  }

  id: number;
  partType: PartType;
  name: string;
  description: string;
}
